pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,400 @@
+	clientauthentication "k8s.io/client-go/pkg/apis/clientauthentication"
+	"k8s.io/client-go/tools/clientcmd/api"
+
+	gardenClient "github.com/gardener/gardenctl-v2/internal/gardenclient"
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+)
+
+// wrappers used for unit tests only
+var (
+	timeNow        = time.Now
+	getCurrentUser = func(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) (string, error) {
+		baseDir, err := api.MakeAbs(path.Dir(authInfo.LocationOfOrigin), "")
+		if err != nil {
+			return "", fmt.Errorf("Could not parse location of kubeconfig origin")
+		}
+
+		if len(authInfo.ClientCertificateData) == 0 && len(authInfo.ClientCertificate) > 0 {
+			err := api.FlattenContent(&authInfo.ClientCertificate, &authInfo.ClientCertificateData, baseDir)
+			if err != nil {
+				return "", err
+			}
+		} else if len(authInfo.Token) == 0 && len(authInfo.TokenFile) > 0 {
+			var tmpValue = []byte{}
+			err := api.FlattenContent(&authInfo.TokenFile, &tmpValue, baseDir)
+			if err != nil {
+				return "", err
+			}
+			authInfo.Token = string(tmpValue)
+		} else if authInfo.Exec != nil && len(authInfo.Exec.Command) > 0 {
+			// The command originates from the users kubeconfig and is also executed when using kubectl.
+			// So it should be safe to execute it here as well.
+			execCmd := exec.Command(authInfo.Exec.Command, authInfo.Exec.Args...)
+			var out bytes.Buffer
+			execCmd.Stdout = &out
+
+			err := execCmd.Run()
+			if err != nil {
+				return "", err
+			}
+
+			var execCredential clientauthentication.ExecCredential
+			err = json.Unmarshal(out.Bytes(), &execCredential)
+			if err != nil {
+				return "", err
+			}
+
+			if token := execCredential.Status.Token; len(token) > 0 {
+				authInfo.Token = token
+			} else if cert := execCredential.Status.ClientCertificateData; len(cert) > 0 {
+				authInfo.ClientCertificateData = []byte(cert)
+			}
+		}
+
+		if len(authInfo.ClientCertificateData) > 0 {
+			block, _ := pem.Decode(authInfo.ClientCertificateData) // does not return an error, just nil
+			if block == nil {
+				return "", fmt.Errorf("Could not decode PEM certificate")
+			}
+
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return "", err
+			}
+
+			user := cert.Subject.CommonName
+			if len(user) > 0 {
+				return user, nil
+			}
+		}
+
+		if len(authInfo.Token) > 0 {
+			tokenReview, err := gardenClient.CreateTokenReview(ctx, authInfo.Token)
+			if err != nil {
+				return "", err
+			}
+			if user := tokenReview.Status.User.Username; user != "" {
+				return user, nil
+			}
+		}
+
+		return "", fmt.Errorf("Could not detect current user")
+	}
+)
+
+//nolint:revive
+type SSHPatchOptions struct {
+	sshBaseOptions
+
+	// BastionName is the patch targets name
+	BastionName string
+	// Factory gives access to the gardenclient and target manager
+	Factory util.Factory
```

Comment:
I think we should not store the factory on options. We have to fix the Run method later.

Reply:
I suggested:

> [petersutter](https://github.com/petersutter) [7 days ago](https://github.com/gardener/gardenctl-v2/pull/180#discussion_r1016693643)
> In the long run do not want to pass the factory anymore to the Run function, hence for new commands we shouldn't use it anymore.
> Instead, put everything you need for the Run function on the options struct.

https://github.com/gardener/gardenctl-v2/pull/180#discussion_r1016693643
but I didn't mean the factory itself :) maybe this was a bit misleading